pygrametl/tables.py

```diff
@@ -1321,6 +1352,118 @@ def closecurrent(self, row, namemapping={}, end=pygrametl.today()):
         if existing[self.toatt] == self.maxto:
             self.update({self.key: keyval, self.toatt: end})
 
+    def _getversions(self, row, namemapping):
```

Comment:
Later, we may add support for doing sorting in the Python code (when `useorderby == False`). I think it will be easier to just handle that in one place and continue to let `_lookuposufusing*` make the assumptions they make now.

About `_` or `__`, I'm not sure what I prefer. I like with `_` that it is easy to override the method to customize the behavior and it is not "dangerous" to call the method so it does not have to be private. On the other hand, it is not clear if anyone else ever would need to a list of version IDs so perhaps it should have two underscores.

I moved it below `lookupasof`

Reply:
> Later, we may add support for doing sorting in the Python code (when `useorderby == False`). I think it will be easier to just handle that in one place and continue to let `_lookuposufusing*` make the assumptions they make now.

I do not understand how that would change if the contents of `_getversions()` is moved to the start of `lookupasof()` as it would still leave a single location for the code but remove the overhead of the method definition for such a short method?

Also I do not understand the benefit of duplicating `versions = self._getversions(row, namemapping)` across all of the `_lookuposufusing*()` methods as is currently done instead of having a single call to `_getversions()` in `lookupasof()` and then passing `versions` to the `_lookuposufusing*()` methods?

Reply:
I think we eventually could get more logic in `_getversions` and I think it would be good if `lookupasofusing...` can be overriden to do something user-defined where the `row` would be needed. The extra method call to `_getversions` has a small cost, but I think it is negligible in the big picture. So I kept it like this for now.